.github/workflows/cli.yml

```diff
@@ -0,0 +1,70 @@
+name: Viam CLI
+
+on:
+  workflow_dispatch:
+    inputs:
+      release_type:
+        required: true
+        type: string
+  workflow_call:
+    inputs:
+      release_type:
+        required: true
+        type: string
+    secrets:
+      GCP_CREDENTIALS:
+        required: true
+
+jobs:
+  viam-cli:
+    strategy:
+      matrix:
+        include:
+        - goos: linux
+          goarch: amd64
+        - goos: linux
+          goarch: arm64
+        - goos: darwin
+          goarch: amd64
+        - goos: darwin
+          goarch: arm64
```

Comment:
I'm not sure that matrix is the right approach here. This will spin up 6 runners, each of which will have to checkout the code, install go, build, and upload. Since you're just changing GOOS/GOARCH, this could be one job on one runner, and just build all 6 binaries there, then upload them together. Should be much more efficient that way.